### PR TITLE
Trying to create search box

### PR DESCRIPTION
**Pull request by greeshma-aeligeti**

### CODE REVIEW

File: EmployeeDirectory.UI/Components/Pages/SearchableDropdown.razor [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        // Delay hiding to allow interaction with dropdown items
+        await Task.Delay(150);
+        isDropdownVisible = false;
+    }
+
+    private object GetPropertyValue(object obj, string propertyName)
+    {
+        if (obj == null)
+        {
+            throw new ArgumentNullException(nameof(obj), "The object cannot be null.");
```
Comment (selvarajjagan) [inline EmployeeDirectory.UI/Components/Pages/SearchableDropdown.razor]: Where are we handling this exception?

File: EmployeeDirectory.UI/Components/Pages/SearchBox.razor [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        searchText = e.Value.ToString();
+        ApplyFiltersAndSort();
+        await OnSearchChanged.InvokeAsync(filteredTasks.ToList());
+    }
+
+
+
+
+    private void ApplyFiltersAndSort()
```
Comment (selvarajjagan) [inline EmployeeDirectory.UI/Components/Pages/SearchBox.razor]: where are we doing Sort?

File: EmployeeDirectory.UI/Components/Pages/SearchableDropdown.razor (hunk 1 above)
Comment (greeshma-aeligeti) [reply EmployeeDirectory.UI/Components/Pages/SearchableDropdown.razor]: I'll handle it today.

File: EmployeeDirectory.UI/Components/Pages/SearchBox.razor (hunk 2 above)
Comment (greeshma-aeligeti) [reply EmployeeDirectory.UI/Components/Pages/SearchBox.razor]: Actually this component is taken from previous task, I thought to remove the unnecessary functions once it starts working fine.

**Pull request merged**